xds/internal/balancer/edsbalancer/balancergroup.go

```diff
@@ -56,29 +109,74 @@ type pickerState struct {
 //     - sub-pickers are grouped into a group-picker
 //     - aggregated connectivity state is the overall state of all pickers.
 //  - resolveNow
+//
+// Sub-balancers are only built when the balancer group is started. If the
+// balancer group is closed, the sub-balancers are also closed. And it's
+// guaranteed that no updates will be sent to parent ClientConn from a closed
+// balancer group.
 type balancerGroup struct {
-	cc balancer.ClientConn
-
-	mu           sync.Mutex
-	idToBalancer map[internal.Locality]balancer.Balancer
-	scToID       map[balancer.SubConn]internal.Locality
-	loadStore    lrs.Store
+	cc        balancer.ClientConn
+	loadStore lrs.Store
 
-	pickerMu sync.Mutex
-	// All balancer IDs exist as keys in this map. If an ID is not in map, it's
-	// either removed or never added.
+	outgoingMu         sync.Mutex
+	outgoingStarted    bool
```

Comment:
I see that all of `outgoingStarted`, `incomingStarted`, `pickerStarted` are set and unset at the same place. Do we really need three different booleans to keep track of this state?